metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/bolt/ParserBolt.java

```diff
@@ -116,6 +122,9 @@ public void execute(Tuple tuple) {
       boolean ackTuple = !writer.handleAck();
       int numWritten = 0;
       if(sensorParserConfig != null) {
+        if (configUpdatedFlag.getAndSet(false)) {
+          parser.configurationUpdated(getSensorParserConfig().getParserConfig());
```

Comment:
You're right that's definitely incorrect.  I think it would be safer if we use the config object fetched in line 118 though.  Is there a benefit to updating the config in the middle of the execute method?  We're already delaying the change in the parser until the next execute.